pkg/controller/servicebindingrequest/binder_test.go

```diff
@@ -145,16 +216,7 @@ func TestBinderNew(t *testing.T) {
 		require.NoError(t, err)
 		require.Len(t, updatedObjects, 1)
 
-		// make sure SBR annonation is added
-		deployment := appsv1.Deployment{}
-		err = runtime.DefaultUnstructuredConverter.FromUnstructured(updatedObjects[0].Object, &deployment)
-		require.NoError(t, err)
-
-		sbrName, err := getSBRNamespacedNameFromObject(&deployment)
-		require.NoError(t, err)
-		require.Equal(t, types.NamespacedName{Name: name, Namespace: ns}, sbrName)
-
-		containers, found, err := unstructured.NestedSlice(updatedObjects[0].Object, containersPath...)
+		containers, found, err := unstructured.NestedSlice(list.Items[0].Object, binder.getContainersPath()...)
```

Comment:
I disagree with this change; this works by coincidence since the list given in `binder.update()` above is being changed in place.

If anything, `list.Items[0]` should be replaced by `updatedObjects`.

Reply:
Regarding the removed tests, I won't argue whether they're good or not but this work doesn't have anything to deal with the topic of this PR it seems.

Reply:
I have put back the removed test. But I had to change one of the assertion.